kolibri/plugins/coach/assets/src/routes/index.js

```diff
@@ -13,6 +13,18 @@ import reportRoutes from './reportRoutes';
 import planRoutes from './planRoutes';
 import { classIdParamRequiredGuard } from './utils';
 
+async function showHomePage(toRoute) {
+  console.log('Inside showHomePage');
+  const initClassInfoPromise = store.dispatch('initClassInfo', toRoute.params.classId);
+  const getFacilitiesPromise =
+    store.getters.isSuperuser && store.state.core.facilities.length === 0
+      ? store.dispatch('getFacilities').catch(() => {})
+      : Promise.resolve();
+
+  await Promise.all([initClassInfoPromise, getFacilitiesPromise]);
+  store.dispatch('notLoading');
```

Comment:
`notLoading` is now called twice for each of the routes being refactored - I'd suggest to remove it from here, or from lines 78 and 89.

Reply:
Resolved.